minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Utils/MineRLTypeHelper.java

```diff
@@ -0,0 +1,135 @@
+     * @return The lowest inventory index matching the contraints, or null if no such slot is found.
+     */
+    @Nullable
+    public static Integer inventoryIndexOf(InventoryPlayer inventory, String itemType, @Nullable Integer metadata) {
+        Item parsedItem = Item.getByNameOrId(itemType);
+        if (parsedItem == null || parsedItem.getRegistryName() == null)
+            throw new IllegalArgumentException(itemType);
+        ResourceLocation targetRegName = parsedItem.getRegistryName();
+
+        if (metadata != null) {
+            validateMetadata(metadata);
+        }
+
+        for (int i = 0; i < inventory.getSizeInventory(); i++) {
+            ItemStack stack = inventory.getStackInSlot(i);
+            ResourceLocation regName = stack.getItem().getRegistryName();
+            if (regName == null) {
+                System.out.printf("null RegistryName at inventory index %d%n", i);
+                return null;
+            }
+
+            boolean flagItemTypeMatches = regName.equals(targetRegName);
+            boolean flagItemMetadataMatches = (metadata == null) || (metadata == stack.getMetadata());
+
+            if (flagItemTypeMatches && flagItemMetadataMatches) {
+                return i;
+            }
+        }
+        return null;
+    }
+
+    public static class ItemTypeMetadataMessage implements IMessage {
+        private String parameters;
+        private String itemType;
+        private Integer metadata;
+
+        public ItemTypeMetadataMessage(){}
+
+        public ItemTypeMetadataMessage(String parameters) {
+            setParameters(parameters);
+        }
+
+        private void setParameters(String parameters) {
+            this.parameters = parameters;
+
+            String[] parts = parameters.split("#");
+            if (parts.length == 1) {
+                itemType = parts[0];
+                metadata = null;
+            } else if (parts.length == 2) {
+                itemType = parts[0];
+                assert parts[0].length() > 0;
+                metadata = Integer.parseInt(parts[1]);
+            } else {
+                throw new IllegalArgumentException(String.format("Bad parameter: '%s'", parameters));
+            }
+            validateMetadata(metadata);
+        }
+
+        public String getItemType() {
+            return itemType;
+        }
+
+        public Integer getMetadata() {
+            return metadata;
+        }
+
+        public boolean validateItemType() {
+            // Sometimes we intentionally send invalid itemType from MineRL -- e.g. "other". In these cases we
+            // should just drop the packet. Ideally, we would error on all unexpected cases... (excluding
+            // "other" and "none"). For now, there seems to be the default behavior of explicitly ignoring "none".
+            Item item = Item.getByNameOrId(getItemType());
+            return item != null && item.getRegistryName() != null && !getItemType().equalsIgnoreCase("none");
```

Comment:
Here we are explicitly ignoring other, or rather treating it as Invalid, which perhaps it is for a command

Reply:
Third clause is not needed but ok to leave